src/main/java/hudson/plugins/gradle/injection/GradleEnterpriseErrorsAction.java

```diff
@@ -0,0 +1,108 @@
+package hudson.plugins.gradle.injection;
+
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.model.RootAction;
+import hudson.model.Run;
+import hudson.plugins.gradle.AbstractBuildScanAction;
+import hudson.util.RunList;
+import jenkins.model.Jenkins;
+import org.kohsuke.stapler.StaplerProxy;
+
+import java.util.Date;
+import java.util.Iterator;
+
+@Extension
+public class GradleEnterpriseErrorsAction implements RootAction, StaplerProxy {
+    @Override
+    public String getIconFileName() {
+        return isVisible() && Jenkins.get().hasPermission(Jenkins.ADMINISTER) ? "/plugin/gradle/images/svgs/gradle-build-scan.svg" : null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Gradle Enterprise";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "gradle_enterprise";
+    }
+
+    @Override
+    public Object getTarget() {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        return isVisible() ? this : null;
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public Iterator<GeErrorModel> getErrors() {
+        return RunList.fromJobs((Iterable) Jenkins.get().allItems(Job.class))
+            .limit(200)
+            .completedOnly()
+            .stream()
+            .filter(r -> {
+                AbstractBuildScanAction action = ((Run) r).getAction(AbstractBuildScanAction.class);
+                return action != null && (action.getHasGradleErrors() || action.getHasMavenErrors());
+            })
+            .map(r -> GeErrorModel.fromRun((Run) r))
+            .iterator();
```

Comment:
Is the idea to show at most 200 errors? In that case I'd filter first and then limit later, as now out of 200 taken, there might be no errors at all even though there were some

Reply:
Good catch 👍 !